lib/index.ts

```diff
@@ -0,0 +1,50 @@
+import { clientInfo } from './environment'
+
+declare var PercyAgent: any;
+
+/**
+ * A function to take a Percy snapshot from a Puppeteer test or script. To use in your tests:
+ *   const { percySnapshot } = require('@percy/puppeteer')
+ *
+ *   const browser = await puppeteer.launch()
+ *   const page = await browser.newPage()
+ *   await page.goto(<your.test.url>)
+ *   await percySnapshot(page, <your snapshot name>, <maybe options>)
+ *
+ * @param page Puppeteer Page object that we are snapshotting. Required.
+ * @param name Name of the snapshot that we're taking. Required.
+ * @param options Additional options, e.g. '{widths: [768, 992, 1200]}'. Optional.
+ */
+export async function percySnapshot(page: any, name: string, options: any = {}) {
+  if (!page) {
+    throw new Error("Puppeteer 'page' object must be provided.")
+  }
+  if (!name) {
+    throw new Error("'name' must be provided. In Mocha, this.test.fullTitle() is a good default.")
+  }
+  await page.addScriptTag({
+    path: require.resolve('@percy/agent/dist/public/percy-agent.js')
+  })
+  await page.evaluate(function(name: string, options: any, clientInfo: string) {
+    const percyAgentClient = new PercyAgent(clientInfo)
```

Comment:
Similarly to the situation with NightmareJS, we don't really have access to info about the Puppeteer environment, and I couldn't find any handy methods on `page` or `page.browser()` to get underlying version or environment information.